sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/TimeWindowUtils.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.feature.management.implementation.timewindow;
+
+import com.azure.spring.cloud.feature.management.implementation.timewindow.recurrence.RecurrenceConstants;
+import org.springframework.util.StringUtils;
+
+import java.time.DayOfWeek;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TimeWindowUtils {
+    public static ZonedDateTime convertStringToDate(String timeStr) {
+        if (!StringUtils.hasText(timeStr)) {
+            return null;
+        }
+        try {
+            return ZonedDateTime.parse(timeStr, DateTimeFormatter.ISO_DATE_TIME);
+        } catch (final DateTimeParseException e) {
+            return ZonedDateTime.parse(timeStr, DateTimeFormatter.RFC_1123_DATE_TIME);
+        }
+    }
+
+    /**
+     * Calculate the number of days that have passed since the start of the week
+     * @param today DayOfWeek enum of today
+     * @param firstDayOfWeek the start day of the week
+     * @return the number of days passed
+     * */
+    public static int daysPassedWeekStart(DayOfWeek today, DayOfWeek firstDayOfWeek) {
```

Comment:
I would suggest considering the .NET FM implementation approach: https://github.com/microsoft/FeatureManagement-Dotnet/blob/main/src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceEvaluator.cs#L337

`int CalculateWeeklyDayOffset(DayOfWeek day1, DayOfWeek day2)`

This method will be a method to *Calculates the offset in days between two given days of the week and returns the number of days to be added to day2 to reach day1*


Since `DayOfWeek` already has the `GetValue` method which returns an `int`, I think `ConvertToWeekDayNumber` is duplicated and also one-liner method is not a good practice.

Reply:
Yes. Remove `convertToWeekDayNumber` method.